### DISCUSSION TITLE
WARNING (drift): It looks like you've created the database CatalogoDatabase multiple times. When these two databases use the same QueryExecutor, race conditions will occur and might corrupt the database

### DISCUSSION THREAD

**Discussion by mgdonato23** [simolus3/drift#3128] [General]
Hello, I need some help. I have two database classes: one called CatalogoMestreDatabase, which controls a main database that receives data at the beginning of my application. Then, I have another class called CatalogoDatabase, which is responsible for managing my other databases. After a certain point in my app, I will receive information in the response.body of an HTTP call to populate other databases that will be created based on the ID. I tried using a Manager to instantiate these databases, as I don't know how many databases I will have in my application since it all depends on the data I receive. However, when I do this, I get the warning: "WARNING (drift): It looks like you've created the database CatalogoDatabase multiple times. When these two databases use the same QueryExecutor, race conditions will occur and might corrupt the database." I wanted to know if there is a way to handle this, please!

**Comment by simolus3** [comment]
> After a certain point in my app, I will receive information in the response.body of an HTTP call to populate other databases that will be created based on the ID

If you have multiple instances of `CatalogoDatabase` that are __all__ pointing to different files, you're fine. As soon as two instances of `CatalogoDatabase` are pointing to the same file, there will be minor problems (stream queries between the two instances won't be synchronized, and since migrations are not typically run in transactions there is a possibility of race conditions around migrations). This is why drift adds that warning - some users were opening multiple instances of the same database files all over the app, and then were surprised to see some features not working.

If that's not the case for you, you can put this snippet somewhere in `main` (or anywhere that runs before opening the databases):

```dart
  driftRuntimeOptions.dontWarnAboutMultipleDatabases = true;
```

That will disable the warning.

**Reply by mgdonato23** [reply]
So I can't use a single file for dynamic database creation? Would I need to create one for each database and not a single one that could do this?

**Reply by simolus3** [reply]
> So I can't use a single file for dynamic database creation

You can do this is a single Dart file / class if that's what you mean.
Dynamic databases can't all be pointing at the same file paths that you pass to `NativeDatabase` though.
